server/src/main/java/org/elasticsearch/common/time/DateFormatters.java

```diff
@@ -178,7 +178,7 @@
     /**
      * Returns a ISO 8601 compatible date time formatter and parser.
      * This is not fully compatible to the existing spec, which would require far more edge cases, but merely compatible with the
-     * existing joda time ISO data formater
+     * existing joda time ISO date formatter
```

Comment:
> but merely compatible with the existing joda time ISO date formatter

I have not tried to go beyond this, but note that the Joda ISO8601 parser _did_ accept timezone for times that didn't include minutes and seconds - see https://www.joda.org/joda-time/apidocs/org/joda/time/format/ISODateTimeFormat.html#timeParser--.  That is why I labelled this PR as `>bug` rather than `>enhancement`.